### PR TITLE
Fix: Allow forward-only on added external models

### PR DESCRIPTION
**Pull request by eakmanrq**
Prior to this change if you added an external model then it would block being able to apply that change as forward-only. Now we only block if the model is an internal model.

### CODE REVIEW

File: sqlmesh/core/context_diff.py [hunk 1]
```diff
@@ -185,6 +185,11 @@ def has_changes(self) -> bool:
     def has_snapshot_changes(self) -> bool:
         return bool(self.added or self.removed or self.modified_snapshots)
 
+    @property
+    def added_internal_models(self) -> t.Set[str]:
+        """Returns the set of added internal models."""
+        return {name for name in self.added if not self.snapshots[name].model_kind_name.is_external}
```
Comment (tobymao) [inline sqlmesh/core/context_diff.py:191]: maybe you should do is_symbolic, or even is_materialized?

File: sqlmesh/core/context_diff.py [hunk 2]
```diff
@@ -185,6 +185,11 @@ def has_changes(self) -> bool:
     def has_snapshot_changes(self) -> bool:
         return bool(self.added or self.removed or self.modified_snapshots)
 
+    @property
+    def added_internal_models(self) -> t.Set[str]:
```
Comment (tobymao) [inline sqlmesh/core/context_diff.py:189]: change the name to represent a change to symbolic or materialized

File: sqlmesh/core/context_diff.py (hunk 1 above)
Comment (izeigerman) [reply sqlmesh/core/context_diff.py:191]: Why? I thought we only needed to treat external models differently.

File: sqlmesh/core/model/common.py [hunk 3]
```diff
@@ -19,7 +19,8 @@ def parse_model_name(name: str) -> t.Tuple[t.Optional[str], t.Optional[str], str
     Returns:
         A tuple consisting of catalog, schema, table name.
     """
-    return split_num_words(name, ".", 3)  # type: ignore
+    table = exp.to_table(name)
+    return table.catalog or None, table.db or None, table.name
```
Comment (tobymao) [inline sqlmesh/core/model/common.py:23]: just table.name

Comment (eakmanrq) [inline sqlmesh/core/model/common.py:23]: I was having an issue with this after fixing the forward-only for external models. Seems like it is time for us to move away from string splitting to get this information. I wasn't sure about dialect here with recent changes related to dialect.

Comment (tobymao) [inline sqlmesh/core/model/common.py:23]: why do you need or None

Comment (tobymao) [reply sqlmesh/core/model/common.py:23]: do we use this anywhere else?

File: sqlmesh/core/snapshot/definition.py [hunk 4]
```diff
@@ -756,7 +756,7 @@ def version_get_or_generate(self) -> str:
     def physical_schema(self) -> str:
         if self.physical_schema_ is not None:
             return self.physical_schema_
-        _, schema, _ = parse_model_name(self.name)
+        schema = parse_model_name(self.name)[1]
```
Comment (tobymao) [inline sqlmesh/core/snapshot/definition.py:759]: i wonder if we just do to_table.schema

File: sqlmesh/core/context_diff.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/context_diff.py:191]: isn’t it more flexible? we have more options with forward only

File: sqlmesh/core/model/common.py (hunk 3 above)
Comment (tobymao) [reply sqlmesh/core/model/common.py:23]: right i forgot to remove this with my refactor

File: sqlmesh/core/snapshot/definition.py (hunk 4 above)
Comment (eakmanrq) [reply sqlmesh/core/snapshot/definition.py]: Good idea going to just replace this with `to_table`

Comment (eakmanrq) [reply sqlmesh/core/snapshot/definition.py]: https://github.com/TobikoData/sqlmesh/pull/1130/commits/79db74c91162a57081296be3f2f2707bf90b4fb2

File: sqlmesh/core/model/common.py (hunk 3 above)
Comment (eakmanrq) [reply sqlmesh/core/model/common.py]: Removed: https://github.com/TobikoData/sqlmesh/pull/1130/commits/79db74c91162a57081296be3f2f2707bf90b4fb2

File: sqlmesh/core/context_diff.py (hunk 1 above)
Comment (eakmanrq) [reply sqlmesh/core/context_diff.py:191]: Yeah it seems like the debate is if embedded or views should be treated like external models in terms of being allowed to be added in forward-only plans.

File: sqlmesh/core/context_diff.py (hunk 2 above)
Comment (eakmanrq) [reply sqlmesh/core/context_diff.py]: https://github.com/TobikoData/sqlmesh/pull/1130/commits/1e8e9505bef66caef0d870612578ea848a844646

File: sqlmesh/core/context_diff.py (hunk 1 above)
Comment (eakmanrq) [reply sqlmesh/core/context_diff.py]: https://github.com/TobikoData/sqlmesh/pull/1130/commits/1e8e9505bef66caef0d870612578ea848a844646

File: sqlmesh/core/snapshot/definition.py [hunk 5]
```diff
@@ -194,8 +194,12 @@ def snapshot_id(self) -> SnapshotId:
 
     @property
     def qualified_view_name(self) -> QualifiedViewName:
-        catalog, schema, table = parse_model_name(self.name)
-        return QualifiedViewName(catalog=catalog, schema_name=schema, table=table)
+        view_name = exp.to_table(self.name)
+        return QualifiedViewName(
+            catalog=view_name.catalog or None,
```
Comment (eakmanrq) [inline sqlmesh/core/snapshot/definition.py:199]: If there isn't a catalog it seems that sqlglot returns an empty string but we are expecting None

**Review by tobymao — APPROVED** [review]

**Pull request merged**